### ISSUE TITLE
"Site.DB" in use by another process

### ISSUE THREAD

**Issue by Kickbut101** [rmountjoy92/DashMachine#141]
Long story short, I rebooted my machine that had version 4 on it (not revision, this was setup 3-11-20), and it no longer ran. I had an error about "thumbs.db" not being found.

Said screw it, copied off fresh files from latest release here on github, ran those, but for some reason running run.py would never work, the error came out like this;

 * Restarting with stat
Traceback (most recent call last):
  File "C:\DashMachine\run.py", line 12, in <module>
    os.remove(db_file_path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\DashMachine\\dashmachine\\user_data\\site.db'

It would however work when I ran the wsgy.py or whatever that was called instead. But I noticed that running that did not allow me to view the dashboard anywhere _but_ on the host machine (by using 127.0.0.1). 

I only managed to get remote access working, and the run.py working by removing lines 11, 12, 13, and 14 from the run.py. Not sure why those are there it looks like some migration thing.

Lines 11-14;
try:
    os.remove(db_file_path)
except FileNotFoundError:
    pass

**Comment by rmountjoy92** [comment]
This sounds like you had another version of DashMachine open in memory that was accessing your user_data folder (and the site.db inside). I'm surprised you didn't get the err: Address already in use message.  Every time DM runs, it deletes the site.db file out of your user_data folder and recreates it using data from the config.

**Comment by Kickbut101** [comment]
I'm not so sure.

1. I only investigated the issue because the dashmachine was not connecting (I have it as homepage on chrome so I noticed right away), and it should have started with a task scheduled command I setup.

2. I used CPorts by sysinternals to kill all applications (python.exe in this case) using port 5000 when I was troubleshooting. So I don't think there was anything still running in the background. And even with this I was still getting that sites.db error.

I will defer to you of course, but I am fairly confident there was nothing else to my knowledge that should have been using that .db file.
